src/protagonist/API.Tests/Integration/CustomHeaderTests.cs

```diff
@@ -0,0 +1,200 @@
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        
+        var model = await response.ReadAsHydraResponseAsync<HydraCollection<CustomHeader>>();
+        model.Members.Should().HaveCount(2);
+    }
+    
+    [Fact]
+    public async Task Post_CustomHeader_201()
+    {
+        // Arrange
+        const int customerId = 94;
+        var path = $"customers/{customerId}/customHeaders";
+        
+        const string newCustomHeaderJson = @"{
+          ""key"": ""test-key"",
+          ""value"": ""test-value"",
+          ""space"": 1,
+        }";
+        
+        // Act
+        var content = new StringContent(newCustomHeaderJson, Encoding.UTF8, "application/json");
+        var response = await httpClient.AsCustomer(customerId).PostAsync(path, content);
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+        
+        var foundCustomHeader = dlcsContext.CustomHeaders.Single(ch => ch.Key == "test-key");
+        foundCustomHeader.Should().NotBeNull();
+        foundCustomHeader.Customer.Should().Be(customerId);
+        foundCustomHeader.Value.Should().Be("test-value");
+        foundCustomHeader.Space.Should().Be(1);
+    }
+    
+    [Fact]
+    public async Task Put_CustomHeader_200()
+    {
+        // Arrange
+        const int customerId = 95;
+        var path = $"customers/{customerId}/customHeaders";
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString(),
+            Customer = customerId,
+            Key = "test-key",
+            Value = "test-value"
+        };
+        const string updatedCustomHeaderJson = @"{
+          ""key"": ""test-key-2"",
+          ""value"": ""test-value-2"",
+          ""space"": 2,
+        }";
+        
+        await dlcsContext.CustomHeaders.AddAsync(customHeader);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var content = new StringContent(updatedCustomHeaderJson, Encoding.UTF8, "application/json");
+        var response = await httpClient.AsCustomer(customerId).PutAsync(Path.Combine(path, customHeader.Id), content);
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+        
+        var updatedCustomHeader = dlcsContext.CustomHeaders.Single(ch => ch.Key == "test-key-2");
+        updatedCustomHeader.Should().NotBeNull();
+        updatedCustomHeader.Value.Should().Be("test-value-2");
+        updatedCustomHeader.Space.Should().Be(2);
+    }
+    
+    [Fact]
+    public async Task Put_CustomHeader_404_IfNotFound()
+    {
+        // Arrange
+        const int customerId = 96;
+        var path = $"customers/{customerId}/customHeaders/{Guid.Empty}";
+        const string updatedCustomHeaderJson = @"{
+          ""key"": ""test-key-2"",
+          ""value"": ""test-value-2"",
+          ""space"": 2,
```

Comment:
what happens when you set this value to a space which doesn't exist? Likely also needs verification that nothing changes on failure

Reply:
https://github.com/dlcs/protagonist/pull/569#discussion_r1280842836